api/facades/action/ActionV7.ts

```diff
@@ -53,7 +53,7 @@ export interface Actions {
 }
 
 export interface ApplicationCharmActionsResult {
-  actions: AdditionalProperties;
+  actions: Record<string, ActionSpec>;
```

Comment:
This is what the type changes from `patternProperties` look like. They all replace `AdditionalProperties` with more specific types (though AdditionalProperties is still used where `additionalProperties` is defined in the schema.